src/sinks/pulsar.rs

```diff
@@ -249,6 +285,12 @@ fn encode_event(mut item: Event, encoding: &EncodingConfig<Encoding>) -> crate::
             .get(log_schema().message_key())
             .map(|v| v.as_bytes().to_vec())
             .unwrap_or_default(),
+        Encoding::Avro => {
+            let value = avro_rs::to_value(log).unwrap();
+            let resolved_value =
+                avro_rs::types::Value::resolve(value, avro_schema.as_ref().unwrap())?;
```

Comment:
Applied.

Reply:
I see a remaining unwrap! Were you unable to remove it?

Reply:
```
error[E0277]: the trait bound `std::option::NoneError: StdError` is not satisfied
   --> src/sinks/pulsar.rs:303:75
    |
303 |                 avro_rs::types::Value::resolve(value, avro_schema.as_ref()?)?;
    |                                                                           ^ the trait `StdError` is not implemented for `std::option::NoneError`
    |
    = note: required because of the requirements on the impl of `From<std::option::NoneError>` for `Box<dyn StdError + std::marker::Send + std::marker::Sync>`
    = note: required by `std::convert::From::from`
```

Not sure I can apply it smoothly, @Hoverbear WDYT?

Reply:
You're right! I'm sorry! I forgot this nuance!